on-target-tests/tests/i2c_loopback.rs

```diff
@@ -0,0 +1,128 @@
+//! This test needs a connection between:
+//!
+//! | from GPIO (pico Pin) | to GPIO (pico Pin) |
+//! | -------------------- | ------------------ |
+//! |         0 (1)        |       2 (4)        |
+//! |         1 (2)        |       3 (5)        |
```

Comment:
Please also mention this in `on-target-tests/README.md`. It already describes the connection necessary for the SPI test, so the I2C connections should be there as well.